osu.Game.Rulesets.Osu/UI/ObjectOrderedHitPolicy.cs

```diff
@@ -22,35 +21,41 @@ public class ObjectOrderedHitPolicy : IHitPolicy
     {
         public IHitObjectContainer HitObjectContainer { get; set; }
 
-        public bool IsHittable(DrawableHitObject hitObject, double time) => enumerateHitObjectsUpTo(hitObject.HitObject.StartTime).All(obj => obj.AllJudged);
-
         public void HandleHit(DrawableHitObject hitObject)
         {
         }
 
-        private IEnumerable<DrawableHitObject> enumerateHitObjectsUpTo(double targetTime)
+        public ClickAction CheckHittable(DrawableHitObject hitObject, double time)
         {
-            foreach (var obj in HitObjectContainer.AliveObjects)
+            int index = HitObjectContainer.AliveObjects.IndexOf(hitObject);
+
+            if (index > 0)
             {
-                if (obj.HitObject.StartTime >= targetTime)
-                    yield break;
+                var previousHitObject = (DrawableOsuHitObject)HitObjectContainer.AliveObjects[index - 1];
+                if (previousHitObject.HitObject.StackHeight > 0 && !previousHitObject.AllJudged)
+                    return ClickAction.Ignore;
```

Comment:
this case right here is why the `ClickAction` abstraction is necessary. There are times when a note *should*, by all other metrics, be shaken, but we don't want it to be (e.g. stacks).